Gems/Atom/RPI/Code/Source/RPI.Public/ViewGroup.cpp

```diff
@@ -0,0 +1,258 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RPI.Public/RPISystemInterface.h>
+#include <Atom/RPI.Public/ViewGroup.h>
+#include <AzCore/Debug/Trace.h>
+
+namespace AZ::RPI
+{
+    ViewGroup::ViewGroup()
+    {
+        m_cameraViews.resize(MaxViewTypes);
+    }
+
+    ViewGroup::~ViewGroup()
+    {
+        m_cameraViews.clear();
+    }
+
+    void ViewGroup::Init(Descriptor desc)
+    {
+        m_desc = desc;
+
+        if (auto rpiSystemInterface = RPISystemInterface::Get())
+        {
+            m_xrSystem = rpiSystemInterface->GetXRSystem();
+            if (m_xrSystem)
+            {
+                m_numSterescopicViews = m_xrSystem->GetNumViews();
+                AZ_Assert(m_numSterescopicViews <= XRMaxNumViews, "Atom only supports two XR views");
+            }
+        }
+
+        for (int i = 0; i < MaxViewTypes; i++)
+        {
+            m_cameraViews[i].m_onProjectionMatrixChangedHandler = MatrixChangedEvent::Handler(
+                [=](const AZ::Matrix4x4& matrix)
+                {
+                    if (m_desc.m_projectionEventFunction)
+                    {
+                        m_desc.m_projectionEventFunction(m_cameraViews[i].m_view);
+                    }
+                    if (m_cameraViews[i].m_projectionMatrixChangedEvent.HasHandlerConnected())
+                    {
+                        m_cameraViews[i].m_projectionMatrixChangedEvent.Signal(matrix);
+                    }   
+                });
+
+            m_cameraViews[i].m_onViewMatrixChangedHandler = MatrixChangedEvent::Handler(
+                [=](const AZ::Matrix4x4& matrix)
+                {
+                    if(m_desc.m_viewEventFunction)
+                    {
+                        m_desc.m_viewEventFunction(m_cameraViews[i].m_view);
+                    }
+                    if (m_cameraViews[i].m_viewMatrixChangedEvent.HasHandlerConnected())
+                    {
+                        m_cameraViews[i].m_viewMatrixChangedEvent.Signal(matrix);
+                    }
+                });
+        }
+    }
+
+    void ViewGroup::CreateMainView(AZ::Name name)
+    { 
+        if (m_cameraViews[DefaultViewType].m_view == nullptr)
+        {
+            m_cameraViews[DefaultViewType].m_view = AZ::RPI::View::CreateView(name, AZ::RPI::View::UsageFlags::UsageCamera);
+        }
+    }
+
+    void ViewGroup::CreateStereoscopicViews(AZ::Name name)
+    {
+        if (m_xrSystem)
+        {
+            for (AZ::u32 i = 0; i < m_numSterescopicViews; i++)
+            {
+                uint32_t xrViewIndex =
+                    i == 0 ? static_cast<uint32_t>(AZ::RPI::ViewType::XrLeft) : static_cast<uint32_t>(AZ::RPI::ViewType::XrRight);
+                AZ::Name xrViewName = AZ::Name(AZStd::string::format("%s XR %i", name.GetCStr(), i));
+                if (m_cameraViews[xrViewIndex].m_view == nullptr)
```

Comment:
The creation of the name can happen inside the condition, that way it creates the name only when it has to.